workflows/cloudify_system_workflows/deployment_update/workflow.py

```diff
@@ -20,6 +20,43 @@
 from .utils import clear_graph
 
 
+def _update_plan_nodes(plan):
+    """Update nodes stored in the plan, to reflect the current instance counts
+
+    The plan will always scaling properties as written in the blueprint,
+    but in reality, the nodes could have been scaled, and there might now
+    be more instances. Update the plan to reflect the currently existing
+    amount of instances.
+    """
+    for node in plan['nodes']:
+        existing_node = workflow_ctx.get_node(node['id'])
+        if not existing_node:
+            continue
+
+        try:
+            scale_properties = node['capabilities']['scalable']['properties']
+        except KeyError:
+            continue
+
+        current_num = existing_node._node.get('number_of_instances')
+        current_planned = \
+            existing_node._node.get('planned_number_of_instances')
+
+        try:
+            # if the node has an `instances: deploy: {num}`, that overrides
+            # any other setting
+            force_deploy = node['instances']['deploy']
```

Comment:
Huh, I would've expected this to have kept the original default (updated by whatever's in the update).